experimental/ir/fdp.go

```diff
@@ -0,0 +1,396 @@
+// Copyright 2020-2025 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ir
+
+import (
+	"slices"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/descriptorpb"
+
+	"github.com/bufbuild/protocompile/experimental/ast/predeclared"
+	"github.com/bufbuild/protocompile/experimental/ast/syntax"
+	"github.com/bufbuild/protocompile/experimental/ir/presence"
+	"github.com/bufbuild/protocompile/experimental/seq"
+	"github.com/bufbuild/protocompile/internal/ext/slicesx"
+)
+
+// DescriptorSet generates a FileDescriptorSet for the given files, and returns the
+// result as an encoded byte slice.
+//
+// The resulting FileDescriptorSet is always fully linked: it contains all dependencies except
+// the WKTs, and all names are fully-qualified.
+func DescriptorSet(files []File, options ...DescriptorOption) ([]byte, error) {
```

Comment:
I'm still trying to understand the rationale for returning bytes. That isn't helpful for _any_ user of this library. They are just going to have to turn around and unmarshal the bytes right back into the proper descriptor proto types. I not really buying the "protobuf-go shouldn't be in the API at all" bit -- it's just unnecessary friction for users, and the value is unclear. Even if we didn't have the runtime dependency on "protobuf-go", I think we'd want something like this -- to bridge to the protobuf-go API, for users to actually make use of the results.

Reply:
Resolved offline; we're gonna just have `*Bytes` variants of these for now.